### PR TITLE
[RUN-277] - Generator flow design has changed

### PR DESCRIPTION
**Pull request by FilyutaDima**
## Description 📝

Generator flow design has changed

Updated the rune selection screen according to the design
Removed the text above rune selection (Собери свой руннй узор …..)
Updated titles on each page of the generator functionality

### Tests ✅

- iPhone 8, iPhone 11 Pro Max, iPhone 14 Pro Max - simulators
- iPhone 11 Pro Max - real device

### CODE REVIEW

**Review by arti1233 — APPROVED** [review]

File: Runar/Runar/Controllers/Generator flow/SelectedRuneCell.swift [hunk 1]
```diff
@@ -83,16 +83,16 @@ class SelectedRuneCell: UICollectionViewCell {
...
+            runeImage.widthAnchor.constraint(equalToConstant: 88),
+            runeImage.heightAnchor.constraint(equalToConstant: 109),
+            runeImage.centerXAnchor.constraint(equalTo: centerXAnchor),
+            runeImage.centerYAnchor.constraint(equalTo: centerYAnchor)
         ])
         
         addSubview(runeName)
         runeName.translatesAutoresizingMaskIntoConstraints = false
         NSLayoutConstraint.activate([
-            runeName.topAnchor.constraint(equalTo: runeImage.bottomAnchor),
```
Comment (SkrigalovskiAndrei) [inline Runar/Runar/Controllers/Generator flow/SelectedRuneCell.swift:95]: Double bottom?
Должно писать о конфликте

File: Runar/Runar/Controllers/Generator flow/SelectionRuneVC.swift [hunk 2]
```diff
@@ -119,12 +108,18 @@ public class SelectionRuneVC: UIViewController, UIGestureRecognizerDelegate {
     }
     
     private func configureNavigationBar() {
-        title = .generateRunesTitle
-        navigationController?.navigationBar.isHidden = false
-        navigationController?.tabBarController?.tabBar.isHidden = false
-        navigationController?.setStatusBar(backgroundColor: .navBarBackground)
+
```
Comment (SkrigalovskiAndrei) [inline Runar/Runar/Controllers/Generator flow/SelectionRuneVC.swift:111]: self здесь не нужен

File: Runar/Runar/Controllers/Generator flow/WallpaperWithBackground/WallpaperWithBackgroundVC.swift [hunk 3]
```diff
@@ -108,7 +108,7 @@ public class WallpaperWithBackgroundVC: UIViewController {
     }
     
     private func configureNavBar() {
-        title = .wallpapersHeader
```
Comment (SkrigalovskiAndrei) [inline Runar/Runar/Controllers/Generator flow/WallpaperWithBackground/WallpaperWithBackgroundVC.swift:111]: extra self

File: Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift [hunk 4]
```diff
@@ -54,3 +59,21 @@ class MainTabBarController: UITabBarController {
         return navigationVC
     }
 }
+
+class CustomIntensityVisualEffectView: UIVisualEffectView {
+    
+    init(effect: UIVisualEffect, intensity: CGFloat) {
+        super.init(effect: nil)
```
Comment (SkrigalovskiAndrei) [inline Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift:66]: Старайся использовать weak вместо unowned

File: Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift [hunk 5]
```diff
@@ -54,3 +59,21 @@ class MainTabBarController: UITabBarController {
...
+        animator = UIViewPropertyAnimator(duration: 1, curve: .linear) { [unowned self] in self.effect = effect }
+        animator.fractionComplete = intensity
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
```
Comment (SkrigalovskiAndrei) [inline Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift:74]: Property размещаются выше инициализатора

Использование "!" очень скользкая дорожка

File: Runar/Runar/Extensions/UINavigationController.swift [hunk 6]
```diff
@@ -36,3 +36,17 @@ extension UINavigationController {
...
+extension UINavigationItem {
+    
+    func setNavigationTitle(_ title: String) {
+        let titleLabel = UILabel (frame: .zero)
+        titleLabel.backgroundColor = .clear
+        titleLabel.textColor = UIColor(red: 0.825, green: 0.77, blue: 0.677, alpha: 1)
+        titleLabel.numberOfLines = 0
+        titleLabel.font = FontFamily.AmaticSC.bold.font(size: 36)
+        titleLabel.textAlignment = .center
+        titleLabel.text = title
```
Comment (SkrigalovskiAndrei) [inline Runar/Runar/Extensions/UINavigationController.swift:49]: extra self

**Review by SkrigalovskyAndrey — CHANGES_REQUESTED** [review]
Fix comments

File: Runar/Runar/Controllers/Generator flow/SelectedRuneCell.swift [hunk 7]
```diff
@@ -83,16 +83,16 @@ class SelectedRuneCell: UICollectionViewCell {
...
+            runeName.bottomAnchor.constraint(equalTo: runeImage.bottomAnchor, constant: 5),
             runeName.bottomAnchor.constraint(equalTo: bottomAnchor),
```
Comment (SkrigalovskyAndrey) [inline Runar/Runar/Controllers/Generator flow/SelectedRuneCell.swift]: Double bottom?
Look like conflict message in console

File: Runar/Runar/Controllers/Generator flow/SelectionRuneVC.swift [hunk 8]
```diff
@@ -119,12 +108,18 @@ public class SelectionRuneVC: UIViewController, UIGestureRecognizerDelegate {
...
-        navigationController?.tabBarController?.tabBar.isHidden = false
-        navigationController?.setStatusBar(backgroundColor: .navBarBackground)
+        
+        self.tabBarController?.tabBar.isHidden = false
+        self.navigationItem.largeTitleDisplayMode = .never
+        self.navigationItem.hidesBackButton = true
+        self.navigationController?.navigationBar.backgroundColor = .clear
         self.navigationController?.navigationBar.configure()
-        navigationItem.largeTitleDisplayMode = .never
+        self.navigationItem.setNavigationTitle(.generateRunesTitle)
```
Comment (SkrigalovskyAndrey) [inline Runar/Runar/Controllers/Generator flow/SelectionRuneVC.swift]: extra self

File: Runar/Runar/Controllers/Generator flow/WallpaperWithBackground/WallpaperWithBackgroundVC.swift [hunk 9]
```diff
@@ -108,7 +108,7 @@ public class WallpaperWithBackgroundVC: UIViewController {
...
+        self.navigationItem.setNavigationTitle(.wallpapersHeader)
         self.navigationItem.hidesBackButton = true
```
Comment (SkrigalovskyAndrey) [inline Runar/Runar/Controllers/Generator flow/WallpaperWithBackground/WallpaperWithBackgroundVC.swift:125]: extra self

File: Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift [hunk 10]
```diff
@@ -54,3 +59,21 @@ class MainTabBarController: UITabBarController {
         return navigationVC
     }
 }
+
+class CustomIntensityVisualEffectView: UIVisualEffectView {
+    
+    init(effect: UIVisualEffect, intensity: CGFloat) {
+        super.init(effect: nil)
+        animator = UIViewPropertyAnimator(duration: 1, curve: .linear) { [unowned self] in self.effect = effect }
```
Comment (SkrigalovskyAndrey) [inline Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift]: Use weak instead unowned

File: Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift [hunk 11]
```diff
@@ -54,3 +59,21 @@ class MainTabBarController: UITabBarController {
...
+        animator.fractionComplete = intensity
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    private var animator: UIViewPropertyAnimator!
```
Comment (SkrigalovskyAndrey) [inline Runar/Runar/Controllers/Misc/TabBarController/MainTabBarController.swift]: Property are placed above the initializer

Usage "!" very slippery road

File: Runar/Runar/Extensions/UINavigationController.swift [hunk 12]
```diff
@@ -36,3 +36,17 @@ extension UINavigationController {
...
+        self.titleView = titleLabel
```
Comment (SkrigalovskyAndrey) [inline Runar/Runar/Extensions/UINavigationController.swift]: extra self

**Comment by Kengii** [comment]
Simulator Iphone SE
1. Background NavBAr
Simulator Screen Shot - Iphone SE - 2023-02-04 at 22 03 23
2. wrong button names, check with figma
Simulator Screen Shot - Iphone SE - 2023-02-04 at 22 04 54
Simulator Screen Shot - Iphone SE - 2023-02-04 at 22 05 03
Simulator Screen Shot - Iphone SE - 2023-02-04 at 22 05 18

**Review by AliaksandrNiehaj — APPROVED** [review]

**Review by SkrigalovskyAndrey — APPROVED** [review]

**Review by Qeezii — APPROVED** [review]

**Review by Kengii — APPROVED** [review]

**Pull request merged**